modules/pam_faillock/pam_faillock.c

```diff
@@ -175,193 +130,18 @@ args_parse(pam_handle_t *pamh, int argc, const char **argv,
 		opts->flags |= FAILLOCK_FLAG_SILENT;
 
 	if (opts->dir == NULL) {
-		pam_syslog(pamh, LOG_CRIT, "Error allocating memory: %m");
-		opts->fatal_error = 1;
+		opts->dir = strdup(FAILLOCK_DEFAULT_TALLYDIR);
+		if (opts->dir == NULL) {
+			pam_syslog(pamh, LOG_CRIT, "Error allocating memory: %m");
+			opts->fatal_error = 1;
+		}
 	}
```

Comment:
Could we avoid unnecessary memory allocations, please?
For example, could we leave `opts->dir == NULL` unchanged and treat it later as `FAILLOCK_DEFAULT_TALLYDIR`?

Reply:
Done